aten/src/ATen/native/Normalization.cpp

```diff
@@ -508,7 +525,30 @@ std::tuple<Tensor, Tensor, Tensor> _batch_norm_impl_index_backward(
   const Tensor& save_mean = c10::value_or_else(save_mean_opt, [] {return Tensor();});
   const Tensor& save_var_transform = c10::value_or_else(save_var_transform_opt, [] {return Tensor();});
 
-  if (impl_index == 0) {
+  if (input.numel()==0) {
+    int64_t num_batch = input.size(0);
+    int64_t num_feature = input.size(1);
+    std::vector<int64_t> dims{0, 2};
+
+    //don't return view of input, don't return empty tensor because it will break gradient chain
+    auto grad_input = grad_output.clone();
+    Tensor grad_weight;
+    Tensor grad_bias;
+    if (output_mask[2]) {
+      grad_bias = grad_input.view({num_batch, num_feature, -1}).sum(dims);
+    }
+    if (output_mask[1]) {
+      grad_weight = (grad_input.view({num_batch, num_feature, -1}) * input).sum(dims);
```

Comment:
I don't think this actually works, as `num_batch` is likely 0, you'll get 
```
In [33]: a=torch.randn(0,2,3,4)

In [34]: a.view(0,2,-1)
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-34-845acd0bbe9f> in <module>
----> 1 a.view(0,2,-1)

RuntimeError: cannot reshape tensor of 0 elements into shape [0, 2, -1] because the unspecified dimension size -1 can be any value and is ambiguous

```
It also means that this is likely untested.

Reply:
Oops, good catch.
I put 0-dim in the wrong spot in my test :facepalm: . https://github.com/pytorch/pytorch/pull/57321/files#diff-4a85700059a0d89291a89a82d2f1ce111293a80c5bc7009b0479fb51b7e5d548R115-R120

Will update.

Reply:
Double oops, I also forgot to put weight/bias on those test cases :facepalm: 
I have no idea what I was doing....